ui/src/singletons/store.ts

```diff
@@ -115,6 +121,11 @@ const useStore = create<AppStore>()(
         createDroppedNode: (eipId, position) =>
           set((state) => {
             const node = newNode(eipId, position)
+            const isHorizontal = state.layout === "horizontal"
+            
+            node.targetPosition = isHorizontal ? Position.Left : Position.Top,
+            node .sourcePosition = isHorizontal ? Position.Right : Position.Bottom
```

Comment:
Can we move this to the `newNode` method? Just for more encapsulation of node creation.